pkg/compact/compact.go

```diff
@@ -1017,12 +1061,12 @@ func (cg *Group) compact(ctx context.Context, dir string, planner Planner, comp
 	toCompactDirs := make([]string, 0, len(toCompact))
 	for _, m := range toCompact {
 		bdir := filepath.Join(dir, m.ULID.String())
-		for _, s := range m.Compaction.Sources {
-			if _, ok := uniqueSources[s]; ok {
-				return false, ulid.ULID{}, halt(errors.Errorf("overlapping sources detected for plan %v", toCompact))
-			}
-			uniqueSources[s] = struct{}{}
-		}
+		//for _, s := range m.Compaction.Sources {
+		//	if _, ok := uniqueSources[s]; ok {
+		//		return false, ulid.ULID{}, halt(errors.Errorf("overlapping sources detected for plan %v", toCompact))
+		//	}
+		//	uniqueSources[s] = struct{}{}
+		//}
```

Comment:
How do we want to proceed with this? I think we still want to keep the logic here if partition compaction is not enabled.